parts/k8s/kubernetesagentcustomdata.yml

```diff
@@ -168,12 +168,13 @@ runcmd:
 - echo `date`,`hostname`, startruncmd>>/opt/m
 # the first arg is the number of retries, the second arg is the wait duration between two retries and the rest of the args are the cmd to run
 - retrycmd_if_failure() { retries=$1; wait=$2; shift && shift; for i in $(seq 1 $retries); do ${@}; [ $? -eq 0  ] && break || sleep $wait; done; echo Executed \"$@\" $i times; }
+- retrycmd_if_success() { retries=$1; wait=$2; shift && shift; for i in $(seq 1 $retries); do ${@}; [ $? -ne 0  ] && break || sleep $wait; done; echo Executed \"$@\" $i times; }
```

Comment:
So this would only retry if the command succeeds? What about if it fails? Why not just run the command several times?

Reply:
This only retries if the command succeeds (such as grep'ing for a failure string).

If the command exits non-zero it should `break` the for loop.

Just running the command several times as well would work assuming the command is idempotent and the most recent run reflects enforced state (in which case always retrying n times where the nth attempt fails would not get you where you want to go). Also, if the command takes time always retrying n times adds T * n to every run, even ones that succeed on the first attempt.